js/core/base-runner.js

```diff
@@ -98,43 +98,49 @@ define(
                 });
                 respecEvents.pub("start", "core/base-runner");
 
+                if (respecConfig.preProcess) {
+                    for (var i = 0; i < respecConfig.preProcess.length; i++) {
+                        try { respecConfig.preProcess[i].apply(this); }
+                        catch (e) { respecEvents.pub("error", e); }
+                    }
+                }
+
+                var pipeline = Promise.resolve();
```

Comment:
We are sure all users of ReSpec are using Promise capable browsers, right? I have not idea if Internet Explorer latest stable supports them... if yes, let's merge this. We can always ask on the mailing list.